src/dmd/errors.d

```diff
@@ -400,6 +400,15 @@ extern (C++) void message(const(char)* format, ...)
     va_end(ap);
 }
 
+/// Alias declaration for delegate to handle diagnostics.
+alias DiagnosticHandler = void delegate(const ref Loc, Color, const(char)*, const(char)*, va_list, const(char)*, const(char)*);
+
+/**
+ * DiagnosticHandler's variable.
+ * Set this to configure what to do with diagnostics (other than printing to console or stderr).
```

Comment:
If think this documentation sounds a bit strange. I would probably say something like:

```
The diagnostic handler.

This will be called for every diagnostic message issued by the compiler. By default it will print the location and the message to stderr.
```